docs: Flex manual, improve SSH instructions

jwwojak opened the pull request:
# Overview

This PR revises and updates Chapter 7, Software, SSH instructions.

- Content reorganization: separate H2s for SSH and Jupyter Notebook (no changes to any Jupyter Notebook content, other than moving to new H2).
- New instructions for hardwired SSH connections.
- New instructions for assigning a static IP address.
- Sections are somewhat separate, any procedural numbering restarts at 1 in each.
- Unrelated suggestion in `mkdocs.yml` in a comment. Recommend setting `toc_depth: 3` to reduce nesting and visual clutter in right side TOC. 

See also JIRA, RTC-696 

## Test Plan and Hands on Testing

The SSH instructions have been tested (multiple times) and work. 

## Changelog

See the overview above.

## Review requests

Please give it a read and share thoughts/recommend changes. Nothing fancy in here. Just markdown.

## Risk assessment

Low for documentation.

jwwojak committed:
Partial revision SSH instructions

Intro paragraph and basic SSH instructions.

Includes SSH key creation and connecting to Flex.

More to come.

jwwojak committed:
Remaining SSH instructions

Adds new hardwired instructions.

Removes old SSH section.

Makes Jupyter Notebook section a new H2 after the new H2 SSH section.

jwwojak committed:
Merge branch 'edge' into improve-flex-ssh-instructions

jwwojak committed:
Merge branch 'edge' into improve-flex-ssh-instructions

jwwojak committed:
Merge branch 'edge' into improve-flex-ssh-instructions

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
 Flex *will not* retain information about more than 20 runs on the robot. Proceeding to the Run Setup screen generates an entry in the list and counts towards the maximum of 20 runs, even if you never begin the protocol.
 
-## Advanced operation
+## Command Line Operation Over SSH
```
Sentence case headers here and throughout.

jwwojak replied:
Changing to sentence case headers

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+## Command Line Operation Over SSH
 
-### Jupyter Notebook
+You can work with your Flex through a Secure Shell (SSH) terminal connection. Terminal access lets you run protocols directly from the command line or perform advanced tasks, such as customizing the Python environment on the robot. Protocols that reference external files on disk (apart from custom labware definition files) must be run from the command line.
```
Let's preserve the link here, even if it goes to a fairly information-light section of the API docs. (We could move or replicate that info here, but I don't think that's necessary for this PR.)
```suggestion
You can work with your Flex through a Secure Shell (SSH) terminal connection. Terminal access lets you [run protocols directly from the command line](https://docs.opentrons.com/v2/new_advanced_running.html#command-line) or perform advanced tasks, such as customizing the Python environment on the robot. Protocols that reference external files on disk (apart from custom labware definition files) must be run from the command line.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
-Access your robot's Jupyter Notebook either:
+!!!Note
+    - SSH keys are required before you can connect to Flex and issue commands from a terminal.
+    - If you're unable to use a Wi-Fi network for SSH, see Hardwired SSH Connections below.
```
Let's link to that section here.
```suggestion
    - If you're unable to use a Wi-Fi network for SSH, see [Hardwired SSH Connections][hardwired-ssh-connections] below.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
-The connection works if you see an ASCII art version of the Opentrons logo. You can now browse the Flex file system and issue commands from the terminal.
+1. Type the passphrase you set when creating the SSH key.
+
+When an SSH connection is successful, the terminal command prompt changes to `root@Flex` followed by the serial number of your robot (e.g., `root@FLXA1020231007001:~#`). You can now interact with the robot via the terminal window.
```
Slight change to keep this consistent, since the example prompt doesn't include the string `Flex` (only the `FLX` that's part of the serial number).
```suggestion
When an SSH connection is successful, the terminal command prompt changes to `root@` followed by the serial number of your robot (e.g., `root@FLXA1020231007001:~#`). You can now interact with the robot via the terminal window.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+A hardwired connection uses an Ethernet cable to connect and transmit data directly between your computer and Flex. This is a secure alternative for SSH access in situations where network policies prevent you from making a wireless connection to the robot.
+
+!!!note
+    The hardwired SSH procedure requires familiarity with working in a terminal window and setting a static IP address. You may want to ask for help from your IT support team before proceeding.
```
I would recommend removing "familiarity with working in a terminal window" since all of the setup instructions above require that too.

jwwojak replied:
Removing quoted text. Will leave in same section. Will use:

> The hardwired SSH procedure requires assigning a static IP address to the robot. You may want to ask for help from your IT support team before proceeding.

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+
+You can get the IP address range and subnet mask from the robot after connecting it to your computer and checking the Opentrons App. These instructions will help you get started:
+
+1. Disconnect the robot's Ethernet cable from the wall jack and connect it to your computer.
```
The first instruction here assumes some odd context. Maybe the Flex isn't connected to anything when they start this! This will integrate these steps better within the section:
```suggestion
You can get the IP address range and subnet mask from the robot by connecting it to your computer and checking the Opentrons App:

1. If the robot is connected by Ethernet cable to a switch or wall jack, disconnect it. Then establish a physical Ethernet connection to your computer, as described above.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+
+1. Launch the Opentrons App.
+
+1. Click the Devices tab and find your robot.
```
```suggestion
1. Click the **Devices** tab and find your robot.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+1. Click the Devices tab and find your robot.
+
+    !!!note
+        If your robot appears as inactive or inaccessible in the app, wait a few moments. Flex will configure itself and eventually become available again. If this does not happen, turn the power off, wait a few seconds, turn the power back on, and check the app again after the robot boots up.
```
Clarifying which device to power cycle.
```suggestion
        If your robot appears as inactive or inaccessible in the app, wait a few moments. Flex will configure itself and eventually become available again. If this does not happen, turn the robot's power off, wait a few seconds, turn the power back on, and check the app again after the robot boots up.
```

ecormany commented on docs/flex-manual/docs/software-operation.md:
```diff
@@ -700,54 +700,110 @@ Each entry in the recent protocol runs list includes the protocol name, its time
+
+#### Physical Connection
+
+Connect a laptop to the robot using an Ethernet cable. If your laptop has a built-in RJ-45 Ethernet port, plug one end into the computer and connect the other end to the Ethernet port on the robot. If you're using a laptop without an Ethernet port, use an adapter with an Ethernet port to make this connection.
```
I think we should replace "laptop" with "computer" throughout. There's no reason you need to use a laptop for this, except the convenience of being able to bring it close to the Flex.

jwwojak replied:
Changing to "computer"

ecormany commented on docs/flex-manual/mkdocs.yml:
Let's not commit this change. Restructuring of the Flex manual is coming.

ecormany reviewed (commented):
Good additions to this section overall. Several suggestions for improved clarity and consistency.

jwwojak committed:
Apply suggestions from code review

Committing suggestions in batch.

Co-authored-by: Ed Cormany <edward.cormany@opentrons.com>

jwwojak commented on docs/flex-manual/mkdocs.yml:
```diff
@@ -48,6 +48,8 @@ markdown_extensions:
       permalink: true
+      # Recommend setting toc_depth: 3 for less clutter and better right-side navigation
+
 
```
Will remove the comment and extra line space

jwwojak reviewed (commented):
Will fix review comments in next push.

jwwojak committed:
Fixing review comments

Change headers to sentence case.

Fix note text in hardwired section.

Replace "laptop" with "computer"

Removed comment and extra line from mkdocs.yml

ecormany reviewed (approved):
Revisions look great. Thank you! :shipit:
